backend/bitboxbase/rpcclient/rpcclient.go

```diff
@@ -213,24 +211,207 @@ func (rpcClient *RPCClient) GetSampleInfo() (rpcmessages.SampleInfoResponse, err
+	}
+	return reply, nil
+}
+
+// EnableTorMiddleware makes an rpc call to BitBoxBase that enables/disables the Tor hidden service for the middleware based on rpcmessages.ToggleSettingEnable/Disable
+func (rpcClient *RPCClient) EnableTorMiddleware(toggleAction rpcmessages.ToggleSetting) (rpcmessages.ErrorResponse, error) {
+	rpcClient.log.Printf("Executing 'EnableTorMiddleware: %v' rpc call", toggleAction)
+	var reply rpcmessages.ErrorResponse
+	err := rpcClient.client.Call("RPCServer.EnableTorMiddleware", toggleAction, &reply)
+	if err != nil {
+		return rpcmessages.ErrorResponse{}, errp.WithStack(err)
+	}
+	return reply, nil
+}
+
+// EnableTorElectrs makes an rpc call to BitBoxBase that enables/disables the Tor hidden service for electrs based on rpcmessages.ToggleSettingEnable/Disable
+func (rpcClient *RPCClient) EnableTorElectrs(toggleAction rpcmessages.ToggleSetting) (rpcmessages.ErrorResponse, error) {
+	rpcClient.log.Printf("Executing 'EnableTorElectrs: %v' rpc call", toggleAction)
+	var reply rpcmessages.ErrorResponse
+	err := rpcClient.client.Call("RPCServer.EnableTorElectrs", toggleAction, &reply)
+	if err != nil {
+		return rpcmessages.ErrorResponse{}, errp.WithStack(err)
+	}
+	return reply, nil
+}
+
+// EnableTorSSH makes an rpc call to BitBoxBase that enables/disables the tor hidden service for SSH based on rpcmessages.ToggleSettingEnable/Disable
+func (rpcClient *RPCClient) EnableTorSSH(toggleAction rpcmessages.ToggleSetting) (rpcmessages.ErrorResponse, error) {
+	rpcClient.log.Printf("Executing 'EnableTorSSH: %v' rpc call", toggleAction)
+	var reply rpcmessages.ErrorResponse
+	err := rpcClient.client.Call("RPCServer.EnableTorSSH", toggleAction, &reply)
+	if err != nil {
+		return rpcmessages.ErrorResponse{}, errp.WithStack(err)
+	}
+	return reply, nil
+}
+
+// EnableClearnetIBD makes an rpc call to BitBoxBase that configures bitcoind to run over clearnet while in IBD mode based on rpcmessages.ToggleSettingEnable/Disable
+func (rpcClient *RPCClient) EnableClearnetIBD(toggleAction rpcmessages.ToggleSetting) (rpcmessages.ErrorResponse, error) {
+	rpcClient.log.Printf("Executing 'EnableClearnetIBD: %v' rpc call", toggleAction)
+	var reply rpcmessages.ErrorResponse
+	err := rpcClient.client.Call("RPCServer.EnableClearnetIBD", toggleAction, &reply)
+	if err != nil {
+		return rpcmessages.ErrorResponse{}, errp.WithStack(err)
+	}
+	return reply, nil
+}
+
+// EnableRootLogin makes an rpc call to BitBoxBase that enables/disables login via the root user/password based on rpcmessages.ToggleSettingEnable/Disable
+func (rpcClient *RPCClient) EnableRootLogin(toggleAction rpcmessages.ToggleSetting) (rpcmessages.ErrorResponse, error) {
+	rpcClient.log.Printf("Executing 'EnableRootLogin: %v' rpc call", toggleAction)
+	var reply rpcmessages.ErrorResponse
+	err := rpcClient.client.Call("RPCServer.EnableRootLogin", toggleAction, &reply)
+	if err != nil {
+		return rpcmessages.ErrorResponse{}, errp.WithStack(err)
+	}
+	return reply, nil
```

Comment:
Can we replace these lines with a more compact function, that takes the rpc call name and args as arguments and returns the error object and go error?

Reply:
Yes agreed, let's do it as a part of separate refactoring PR. Filed an issue: https://github.com/shiftdevices/godbb/issues/876